src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/Razor/RazorDynamicFileInfoProvider.TextChangesTextLoader.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.LanguageServer.LanguageServer;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.HostWorkspace.Razor;
+
+internal partial class RazorDynamicFileInfoProvider
+{
+    private sealed class TextChangesTextLoader(
+        TextDocument? document,
+        IEnumerable<TextChange> changes,
+        byte[] checksum,
+        SourceHashAlgorithm checksumAlgorithm,
+        string? encoding,
+        Uri razorUri) : TextLoader
+    {
+        public override async Task<TextAndVersion> LoadTextAndVersionAsync(LoadTextOptions options, CancellationToken cancellationToken)
+        {
+            if (document is null)
+            {
+                var text = SourceText.From("").WithChanges(changes);
```

Comment:
Does this need to specify the checksum algo and encoding we got from the request?

Reply:
yes it does